src/App.js

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
+import Navigation from './routes/Navigation';
+import BookList from './components/BookList';
+import BookForm from './components/BookForm';
+
+const App = () => {
+  const [books, setBooks] = React.useState([]);
+
+  const handleAddBook = (newBook) => {
+    setBooks((prevBooks) => [...prevBooks, newBook]);
+  };
+
+  const handleDeleteBook = (bookId) => {
+    setBooks((prevBooks) => prevBooks.filter((book) => book.id !== bookId));
+  };
+
+  return (
+    <Router>
+      <div>
+        <Navigation />
+        <Routes>
+          <Route path="/" element={<BookList books={books} onDelete={handleDeleteBook} />} />
+          <Route path="/components" element={<BookForm />} />
+        </Routes>
```

Comment:
- [ ] Good job so far implementing the changes requested by the previous reviewer, however, as he mentioned and requested by the requirement, `Add 2 routes in your application, each should render a page component:`
`/`
`/categories`

As indicated above, routes should include Home and Categories. Please resolve that. :slightly_smiling_face: